ignite/cmd/chain.go

```diff
@@ -60,13 +72,79 @@ chain.
 		Args:    cobra.ExactArgs(1),
 	}
 
-	c.AddCommand(
-		NewChainServe(),
-		NewChainBuild(),
-		NewChainInit(),
-		NewChainFaucet(),
-		NewChainSimulate(),
-	)
+	c.AddCommand(addConfigMigrationVerifier(NewChainServe()))
+	c.AddCommand(addConfigMigrationVerifier(NewChainBuild()))
+	c.AddCommand(addConfigMigrationVerifier(NewChainInit()))
+	c.AddCommand(addConfigMigrationVerifier(NewChainFaucet()))
+	c.AddCommand(addConfigMigrationVerifier(NewChainSimulate()))
```

Comment:
@jeronimoalbi An alternative to prefix all sub commands is to set `PersistentPreRunE` in the parent command.

```go
    // PersistentPreRun: children of this command will inherit and execute.
	PersistentPreRun func(cmd *Command, args []string)
```

I tested it and it's only run when a sub-command is invoked (so not on `ignite chain`), and only if the sub-commands conditions are met (number of args).

This simplifies the code in `addConfigMigrationVerifier` (no need to check if the sub-command has already a `PreRun` function defined) and is also less verbose :)

Reply:
Nice improvement! I didn't realise that it could be refactor to `PersistentPreRun`.

Reply:
Changed in ede171be9c3a6951489c2206944b4bf483f85b80